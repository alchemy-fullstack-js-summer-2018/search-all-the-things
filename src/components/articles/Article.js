import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Article extends Component {

    static propTypes = {
      article: PropTypes.object.isRequired
    };

    render() {
      const { article } = this.props;
      const { Title, Year, Poster } = article;

      return (
        <li>
          <h2><strong>{Title}</strong> ({Year})</h2> 
          <img src={Poster}/>
        </li>
      );
    }
}