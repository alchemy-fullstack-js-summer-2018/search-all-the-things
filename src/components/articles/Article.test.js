import React from 'react';
import { shallow } from 'enzyme';
import Article from './Article';
import toJSON from 'enzyme-to-json';

const article = {
  'source': {
    'id': null,
    'name': 'pdx.com'
  },
  'author': 'that guy',
  'title': 'some text',
  'description': 'some description',
  'url': 'www.pdx.com',
  'urlToImage': 'picture.jpg',
  'publishedAt': '2018-08-15T21:00:04Z'
};

describe('Article', () => {
  
  it('renders correctly', () => {
    const wrapper = shallow(<Article article={article}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders without image', () => {
    const noImageArticle = { ...article, urlToImage: null };
    const wrapper = shallow(<Article article={noImageArticle}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});