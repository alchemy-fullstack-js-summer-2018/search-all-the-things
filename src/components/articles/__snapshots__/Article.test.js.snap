// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Article test Renders as designed 1`] = `
<li>
  <h2>
    2018 FORD MUSTANG COBRA JET FIRST LOOK: ONE FAST PONY
  </h2>
  <p>
    by 
    Alisa Priddle 
     on 
    2018-08-15T21:00:04Z
  </p>
  <a
    href="https://www.motortrend.com/cars/ford/mustang/2018/2018-ford-mustang-cobra-jet-first-look-review/"
    rel="noopener noreferrer"
    target="_blank"
  >
    view article
  </a>
  <p>
    Only 68 of the $130,000 race cars will be made
  </p>
  <img
    src="http://fordauthority.com/wp-content/uploads/2015/12/Ford-Mustang-Cobra-Jet-for-sale-720x340.jpg"
  />
</li>
`;

exports[`Single Article test Renders without image 1`] = `
<li>
  <h2>
    2018 FORD MUSTANG COBRA JET FIRST LOOK: ONE FAST PONY
  </h2>
  <p>
    by 
    Alisa Priddle 
     on 
    2018-08-15T21:00:04Z
  </p>
  <a
    href="https://www.motortrend.com/cars/ford/mustang/2018/2018-ford-mustang-cobra-jet-first-look-review/"
    rel="noopener noreferrer"
    target="_blank"
  >
    view article
  </a>
  <p>
    Only 68 of the $130,000 race cars will be made
  </p>
  <img
    src={null}
  />
</li>
`;
